adr/minimal-sdk.md

```diff
@@ -0,0 +1,53 @@
+* **Reference Implementation:** It will act as the reference for how to correctly interact with the Llama Stack Responses API in an agentic fashion, providing a foundation that other tools can build upon or refer to.  It may be useful as a reference for developers who want to build their own Agent abstraction for a single application.
+
+This two-pronged approach (a minimal "first-step" SDK and a clear "next-step" recommendation of powerful external tools) provides the best balance of speed, user choice, and focused engineering effort.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+* **Rapid Time-to-Market:** A minimal SDK is a fast way to deliver a functional agent construct to our users, allowing us to provide a complete end-to-end story quickly.
+* **Empowers User Choice:** This strategy explicitly embraces the rich AI ecosystem. Customers with existing expertise in established frameworks can use their preferred tools, while new users have a simple, non-intimidating entry point.
+* **Low Engineering Overhead:** By strictly limiting the scope of our own SDK, we avoid the massive maintenance burden of creating and supporting a full-featured framework. Engineering efforts are focused and contained.
+* **Simplified Tooling:** Having a single, simple, Red Hat-owned SDK as a target greatly simplifies the development of any integrated tooling (e.g., agent creation wizards in the RHOAI console).
+* **Avoids Containerization Complexity:** Choosing an SDK over a container image avoids the overhead of building, distributing, securing, and running a container for what is fundamentally client-side logic, simplifying the developer workflow.
+
+### Negative
+
+* **Weakens the "Unified Framework" Narrative:** This approach makes it harder to market Llama Stack or a Red Hat offering as a single, all-in-one agent framework. The story is more nuanced ("use our simple SDK to start, then use these other tools").
+* **Risk of SDK Fragmentation and Confusion:** We must provide clear documentation and guidance to help users understand when to use the minimal Red Hat SDK and when it's time to "graduate" to a more powerful framework.
+* **Limited to Python:** The decision to write the SDK only in Python will limit its direct usefulness for customers who prefer other languages. It also presents a challenge for our internal tooling teams who may use other languages like Go. We accept these trade-offs, acknowledging that most of the target AI/ML developer audience works primarily in Python and that workarounds can be found for internal tooling.
+* **Potential for an Inadequate SDK:** There is a risk that our minimal SDK will be too limited even for slightly complex use cases, forcing a migration to another framework very early in the development process. The initial scope must be defined carefully to be useful.
```

Comment:
I think the scope should be defined before commiting to this path.

Reply:
Yes, I think the lack of clear scope is the current issue.